frontend/cypress/e2e/WEB-customer-list.cy.js

```diff
@@ -0,0 +1,45 @@
+describe('EngageSphere Home', () => {
+    const sizes = [
+        'All',
+        'Small',
+        'Medium',
+        'Enterprise',
+        'Large Enterprise',
+        'Very Large Enterprise'
+    ];
+
+    beforeEach(() => {
+        cy.visit('/');
+    });
+
+    it('Should render the header with an h11', () => {
+        cy.contains('h1', 'EngageSphere').should('be.visible');
+    });
+
+    it('Should toggle theme when toggle button is clicked', () => {
+        cy.get('#theme-toggle-button').should('be.visible').click();
+        cy.get('body').should('have.attr', 'data-theme', 'dark');
+
+    });
+    it('Must have all size options available in the filter and filter correctly', () => {
+        cy.get('[data-testid="filter"]').find('option').then(options => {
+            const currentSizes = [...options].map(option => option.value);
+            expect(currentSizes).to.deep.equal(sizes);
+        });
+
+        sizes.forEach(size => {
+            cy.get('[data-testid="filter"]').select(size).should('have.value', size);
+            cy.get('body').then(($body) => {
+                if ($body.find('.customer').length > 0) {
```

Comment:
Loops and conditionals should not be used in automated tests in 99.99% of cases, and this test should not need such complexities.

I'm just going to ask you. Who tests such complex logic you have implemented?

Reply:
Below is my implementation.

```js
context('Filtering', () => {
    it('filters by All sizes', () => {
      cy.intercept(
        'GET',
        `${Cypress.env('API_URL')}/customers?page=1&limit=10&size=Small`,
        { fixture: 'smallCustomers' }
      ).as('getSmallCustomers')
      // First, filter by a different size (e.g., Small)
      // So that when filtering by All, the `getCustomers` request happens again,
      // and the test can wait for it.
      cy.get('[data-testid="filter"]').select('Small')
      cy.wait('@getSmallCustomers')
      cy.get('[data-testid="filter"]').select('All')
      cy.wait('@getCustomers')

      cy.get('tbody tr').should('have.length', 5)
    })

    it('filters by Small size', () => {
      cy.intercept(
        'GET',
        `${Cypress.env('API_URL')}/customers?page=1&limit=10&size=Small`,
        { fixture: 'smallCustomers' }
      ).as('getSmallCustomers')

      cy.get('[data-testid="filter"]').select('Small')

      cy.wait('@getSmallCustomers')

      cy.get('tbody tr').should('have.length', 1)
    })

    it('filters by Medium size', () => {
      cy.intercept(
        'GET',
        `${Cypress.env('API_URL')}/customers?page=1&limit=10&size=Medium`,
        { fixture: 'mediumCustomers' }
      ).as('getMediumCustomers')

      cy.get('[data-testid="filter"]').select('Medium')

      cy.wait('@getMediumCustomers')

      cy.get('tbody tr').should('have.length', 1)
    })

    it('filters by Enterprise size', () => {
      cy.intercept(
        'GET',
        `${Cypress.env('API_URL')}/customers?page=1&limit=10&size=Enterprise`,
        { fixture: 'enterpriseCustomers' }
      ).as('getEnterpriseCustomers')

      cy.get('[data-testid="filter"]').select('Enterprise')

      cy.wait('@getEnterpriseCustomers')

      cy.get('tbody tr').should('have.length', 1)
    })

    it('filters by Large Enterprise size', () => {
      cy.intercept(
        'GET',
        `${Cypress.env('API_URL')}/customers?page=1&limit=10&size=Large%20Enterprise`,
        { fixture: 'largeEnterpriseCustomers'}
      ).as('getLargeEnterpriseCustomers')

      cy.get('[data-testid="filter"]').select('Large Enterprise')

      cy.wait('@getLargeEnterpriseCustomers')

      cy.get('tbody tr').should('have.length', 1)
    })

    it('filters by Very Large Enterprise size', () => {
      cy.intercept(
        'GET',
        `${Cypress.env('API_URL')}/customers?page=1&limit=10&size=Very%20Large%20Enterprise`,
        { fixture: 'veryLargeEnterpriseCustomers'}
      ).as('getVeryLargeEnterpriseCustomers')

      cy.get('[data-testid="filter"]').select('Very Large Enterprise')

      cy.wait('@getVeryLargeEnterpriseCustomers')

      cy.get('tbody tr').should('have.length', 1)
    })
})
```

It's much longer, but you read it from top to bottom.
It's linear.
There are no loops, conditionals, etc.

It's just simple.

Simplicity is what a good design is all about.

Prefer longer code that is easier to read, than shorter code that is hard to read.

Reply:
Notice that I use fixtures to make my tests more robust and fast.

I'm only testing to ensure the frontend sends the correct query string to the backend.